dandi/dandiarchive.py

```diff
@@ -474,18 +475,25 @@ def parse(cls, url, *, map_instance=True):
 
     @staticmethod
     def follow_redirect(url):
-        r = requests.head(url, allow_redirects=True)
-        if r.status_code == 404:
-            raise NotFoundError(url)
-        elif r.status_code != 200:
-            raise FailedToConnectError(
-                f"Response for getting {url} to redirect returned {r.status_code}."
-                f" Please verify that it is a URL related to dandiarchive and"
-                f" supported by dandi client"
-            )
-        elif r.url != url:
-            return r.url
-        return url
+        i = 0
+        while True:
```

Comment:
why custom loop, I thought that we have `try_multiple` using `tenacity` for such purposes?

Reply:
`try_multiple()` doesn't support the 0.1, 1, 10 sequence of sleep times that you requested.

Reply:
it could be any reasonable sequence it supports, I do not care about precise timing, just spread it so it tries for up to 10ish seconds

Reply:
I feel that, in this case, the code is simpler without involving `try_multiple()`.

Reply:
ok, I will resort to your judgement, lets proceed, thank you!